plugins/source/azure/codegen/recipes/web.go

```diff
@@ -20,6 +20,21 @@ func Web() []Resource {
 		mockListResult:           mockDirectResponse,
 	}
 
+	var authSettingsV2Resource = resourceDefinition{
+		azureStruct:          &web.SiteAuthSettingsV2{},
+		listFunction:         "GetAuthSettingsV2",
+		listFunctionArgsInit: []string{"site := parent.Item.(web.Site)"},
```

Comment:
Not necessarily related to this PR and I think brought this up in the past. Im not a big fan of this part which generates the fetch/resolver:

```
		listFunctionArgsInit: []string{"site := parent.Item.(web.Site)"},
		listFunctionArgs:     []string{"*site.ResourceGroup", "*site.Name"},
		listHandler: `if err != nil {
				return err
			}
			res <- response`,
		mockListFunctionArgsInit: []string{""},
```

hand-writing the resolver is the same amount of lines of code:

```
func fetchWebSiteAuthSettingsV2(ctx context.Context, meta schema.ClientMeta, parent *schema.Resource, res chan<- interface{}) error {
	svc := meta.(*client.Client).Services().Web.SiteAuthSettingsV2

	site := parent.Item.(web.Site)
	response, err := svc.GetAuthSettingsV2(ctx, *site.ResourceGroup, *site.Name)
	if err != nil {
		return err
	}
	res <- response
	return nil
}
```

but it makes the following easier:
1) making changes with auto-complete and code editor. 
2) development cycle shorter as you don't need to go to the `string` that generates the same amount of code.

Reply:
> hand-writing the resolver is the same amount of lines of code

I think this misses the auto generation of relations, mocks and related hierarchy, specifically:
https://github.com/cloudquery/cloudquery/pull/3269/files#diff-ace9a63acb9a115db549f8e20ad70309187d445c0c5896f0c19e08ed4b8fb019R272
https://github.com/cloudquery/cloudquery/pull/3269/files#diff-fa2265cf35c6f5c185b76162db90ff3747eaf79d53a6c96ad5e470227db68880R31
https://github.com/cloudquery/cloudquery/pull/3269/files#diff-ed8636bbd866046d78e64dcc81ec0e14980f09b272fb0f51257ed4be760ccf2fR17

which are also auto generated.

I do think it need to be improved in a different way. Instead of manually writing resolvers that need custom code via the code gen, we should write that code under the service, e.g https://github.com/cloudquery/cloudquery/pull/3269/files#diff-d8dc88c32f2472a2227edefe1c0e386ffcff5b041ee83ef32f47de17497e77baR38 should be a wrapper on top of the Azure SDK that gets the `Site` then does any custom code inside it.

Then we'll have a separation between autogenerated code and manually written code